File: SpringBootJpa/src/main/java/com/example/java/simple/web/CustomController.java
package com.example.java.simple.web;

import com.example.java.simple.domain.Customer;
import com.example.java.simple.service.CustomService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

@RestController
@RequestMapping("/customer")
public class CustomController {

	@Autowired
	private CustomService service;

	@PostMapping
	public ResponseEntity<?> upsert(@RequestBody Customer customer) {
		return ResponseEntity.ok(service.upsert(customer));
	}

	@GetMapping(value = "/findByName")
	public ResponseEntity<?> findByName(@RequestParam("name") String name) {
		return ResponseEntity.ok(service.findByName(name));
	}

	@GetMapping(value = "/{id}")
	public ResponseEntity<?> find(@PathVariable("id") Long id) {
		return ResponseEntity.ok(service.find(id));
	}
}

File: SpringBootJpa/src/main/kotlin/com/example/kotlin/simple/web/CustomController.kt
package com.example.kotlin.simple.web

import com.example.kotlin.simple.domain.Customer
import com.example.kotlin.simple.service.CustomService
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("/customer")
class CustomController(val service: CustomService) {

    @PostMapping
    fun upsert(@RequestBody customer: Customer): ResponseEntity<*> {
        return ResponseEntity.ok(service.upsert(customer))
    }

    @GetMapping(value = ["/findByName"])
    fun findByName(@RequestParam("name") name: String): ResponseEntity<*> {
        return ResponseEntity.ok(service.findByName(name))
    }

    @GetMapping(value = ["/{id}"])
    fun find(@PathVariable("id") id: Long?): ResponseEntity<*> {
        return ResponseEntity.ok(service.find(id))
    }
}